### PR TITLE
Add timing API for handling XRCE-DDS session

### PR DESCRIPTION
**Pull request by pablogs9**
Before merge: https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/260

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         - name: Download dependencies
           run: |
             git clone -b foxy https://github.com/eProsima/Micro-CDR src/Micro-CDR
-            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
+            git clone -b develop https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
```
Comment (pablogs9) [inline .github/workflows/ci.yml:25]:
```suggestion
            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
```

Comment (pablogs9) [reply .github/workflows/ci.yml:25]: Change this and upgrade `foxy`->`develop` in XRCE client before merge.

File: rmw_microxrcedds_c/src/rmw_microros/timing.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#include <rmw_microxrcedds_c/config.h>
+#include <rmw/rmw.h>
+#include <rmw/allocators.h>
+#include <rmw/ret_types.h>
+#include <rmw/error_handling.h>
+
+#include "../types.h"
+
+void rmw_uros_set_publisher_session_timeout(
```
Comment (Acuadros95) [inline rmw_microxrcedds_c/src/rmw_microros/timing.c:23]:
```suggestion
rmw_ret_t rmw_uros_set_publisher_session_timeout(
```

File: rmw_microxrcedds_c/src/rmw_microros/timing.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  rmw_publisher_t * publisher,
+  int session_timeout)
+{
+  rmw_uxrce_publisher_t * custom_publisher = (rmw_uxrce_publisher_t *)publisher->data;
+
+  custom_publisher->session_timeout = session_timeout;
```
Comment (Acuadros95) [inline rmw_microxrcedds_c/src/rmw_microros/timing.c:29]:
```suggestion
  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
  rmw_uxrce_publisher_t * custom_publisher = (rmw_uxrce_publisher_t *)publisher->data;

  custom_publisher->session_timeout = session_timeout;
  return RMW_RET_OK;
```

File: rmw_microxrcedds_c/src/rmw_microros/timing.c [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+void rmw_uros_set_service_session_timeout(
+  rmw_service_t * service,
+  int session_timeout)
+{
+  rmw_uxrce_service_t * custom_service = (rmw_uxrce_service_t *)service->data;
+
+  custom_service->session_timeout = session_timeout;
+}
```
Comment (Acuadros95) [inline rmw_microxrcedds_c/src/rmw_microros/timing.c:39]:
```suggestion
rmw_ret_t rmw_uros_set_service_session_timeout(
  rmw_service_t * service,
  int session_timeout)
{
  RMW_CHECK_ARGUMENT_FOR_NULL(service, RMW_RET_INVALID_ARGUMENT);
  rmw_uxrce_service_t * custom_service = (rmw_uxrce_service_t *)service->data;

  custom_service->session_timeout = session_timeout;
  return RMW_RET_OK;
}
```

File: rmw_microxrcedds_c/src/rmw_microros/timing.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+void rmw_uros_set_client_session_timeout(
+  rmw_client_t * client,
+  int session_timeout)
+{
+  rmw_uxrce_client_t * custom_client = (rmw_uxrce_client_t *)client->data;
+
+  custom_client->session_timeout = session_timeout;
+}
```
Comment (Acuadros95) [inline rmw_microxrcedds_c/src/rmw_microros/timing.c:48]:
```suggestion
rmw_ret_t rmw_uros_set_client_session_timeout(
  rmw_client_t * client,
  int session_timeout)
{
  RMW_CHECK_ARGUMENT_FOR_NULL(client, RMW_RET_INVALID_ARGUMENT);
  rmw_uxrce_client_t * custom_client = (rmw_uxrce_client_t *)client->data;

  custom_client->session_timeout = session_timeout;
  return RMW_RET_OK;
}
```

File: rmw_microxrcedds_c/include/rmw_microros/timing.h [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  rmw_service_t * service,
+  int session_timeout);
+
+/**
+ * \brief Sets the DDS-XRCE session spin time in reliable service client
+ *
+ * \param[in] client client where the spin time is configured
+ * \param[in] session_timeout time in milliseconds
+ */
+void rmw_uros_set_client_session_timeout(
```
Comment (Acuadros95) [inline rmw_microxrcedds_c/include/rmw_microros/timing.h:63]:
```suggestion
 * \return RMW_RET_OK when success.
 * \return RMW_RET_INVALID_ARGUMENT If client is not valid or unexpected arguments.
 */
rmw_ret_t rmw_uros_set_client_session_timeout(
```

File: rmw_microxrcedds_c/include/rmw_microros/timing.h [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  rmw_publisher_t * publisher,
+  int session_timeout);
+
+/**
+ * \brief Sets the DDS-XRCE session spin time in reliable service server
+ *
+ * \param[in] service service where the spin time is configured
+ * \param[in] session_timeout time in milliseconds
+ */
+void rmw_uros_set_service_session_timeout(
```
Comment (Acuadros95) [inline rmw_microxrcedds_c/include/rmw_microros/timing.h:53]:
```suggestion
 * \return RMW_RET_OK when success.
 * \return RMW_RET_INVALID_ARGUMENT If service is not valid or unexpected arguments.
 */
rmw_ret_t rmw_uros_set_service_session_timeout(
```

File: rmw_microxrcedds_c/include/rmw_microros/timing.h [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+ *  @{
+ */
+
+/**
+ * \brief Sets the DDS-XRCE session spin time in reliable publication
+ *
+ * \param[in] publisher publisher where the spin time is configured
+ * \param[in] session_timeout time in milliseconds
+ */
+void rmw_uros_set_publisher_session_timeout(
```
Comment (Acuadros95) [inline rmw_microxrcedds_c/include/rmw_microros/timing.h:43]:
```suggestion
 * \return RMW_RET_OK when success.
 * \return RMW_RET_INVALID_ARGUMENT If publisher is not valid or unexpected arguments.
 */
rmw_ret_t rmw_uros_set_publisher_session_timeout(
```

**Review by Acuadros95 — COMMENTED** [review]
Add check on arguments

**Review by Acuadros95 — APPROVED** [review]

**Comment by pablogs9** [comment]
@mergify backport foxy galactic

**Pull request merged**